docs/changelog.md

```diff
@@ -4,6 +4,11 @@ Lines for version numbers should always be formatted as `* MAJOR.MINOR.PATCH`
 with nothing else on the line.
 -->
 * HEAD
+    * [improvement] BREAKING CHANGE: The existing settings have been split into specific
```

Comment:
Can we have instructions for users on how to change their existing settings to comply with the new format? Or better yet, something to do it automatically?

Reply:
How about a Python migration script that instead of migrating the DB, migrates the yaml files?

Reply:
that's fine, but we will ideally want it to either (a) run automatically or (b) automatically prompt users to do it. That way people don't miss that this requirement exists and then get frustrated when it breaks.

Reply:
Here's an option that might be cleaner: in this PR, add a `schema_version` to `settings.yaml`. In `versions.py`, check `settings.yaml` for a `schema_version.` If it is not present, raise an exception that says `"You must upgrade your settings files, here's how: ..."` . In the future, if we change anything else about the settings schema we can leverage the same mechanism.

Reply:
@augray: Offline we decided to instead introduce a validation for the config dir when the config dir location function is invoked, validation which can be modified in the future for other use cases.

In this case, the validation would be that if the user settings file exists, the server settings file must also exist, so I introduced this change, printing an error message explaining how to manually migrate the settings, and also expanded the documentation to list all supported user and server settings.

PTAL

cc @neutralino1